### PR TITLE
Add document for how to add a code generated resource

### LINKED ISSUE #1683: Add documentation for how to onboard resources to the generator
There should be documentation that clearly explains how to onboard new resources to the generator. Including things like:

- When and how to prune properties
- Common/likely pitfalls (cross resource references, etc)
- How to add (and record) a new test

In the future we could expand it to cover handcrafted resources or resource customizations as well.

### PR DESCRIPTION
**Pull request by matthchr**
Closes #1683

### CODE REVIEW

File: docs/v2/adding-new-codegenerated-resource.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+```
+
+## Run the code generator
+
+See [the code generator README](/hack/README.md) for how to run the code generator.
+
+## Fix any errors raised by the code generator
+
+TODO: expand on common errors
```
Comment (matthchr) [inline docs/v2/adding-new-codegenerated-resource.md:63]: Should do this some

File: docs/v2/adding-new-codegenerated-resource.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+If the resource is not in either of the above places and cannot be deployed via an ARM template, then it's not an ARM resource and currently cannot be code generated.
+
+## Determine a resource to add
+There are three key pieces of information required before adding a resource to the code generation configuration file, and each of them can be found in the 
+[ARM template JSON schema](https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json) or the [auto-generated ARM template JSON schema](https://schema.management.azure.com/schemas/common/autogeneratedResources.json). To get started, find the entry for the resource in one of the two templates mentioned. For example, the entry for Azure Network Security Groups looks like this:
+`{ "$ref": "https://schema.management.azure.com/schemas/2020-11-01/Microsoft.Network.json#/resourceDefinitions/networkSecurityGroups" },`
+
+**Note**: In many cases there will be multiple entry's for the same resource, each with a different `api-version`. It is _strongly_ recommended that you use the latest available `api-version`.
```
Comment (theunrepentantgeek) [inline docs/v2/adding-new-codegenerated-resource.md:17]:
```suggestion
**Note**: In many cases there will be multiple entry's for the same resource, each with a different `api-version`. It is _strongly_ recommended that you use the latest available non-preview `api-version`.
```

File: docs/v2/adding-new-codegenerated-resource.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+### The `name` of the resource
+You usually know this going in. In our example above, the name of the resource is `networkSecurityGroups`.
+
+### The `group` the resource is in 
+This is usually named after the Azure service, for example `microsoft.resources` or `microsoft.documentdb`. In our example entry from above, this is `microsoft.network`.
+
+### The `api-version` of the resource
+This is usually a date, sometimes with a `-preview` suffix. In our example entry from above, this is `2020-11-01`.
```
Comment (theunrepentantgeek) [inline docs/v2/adding-new-codegenerated-resource.md:26]: Above it reads

> There are three key pieces of information required

So maybe use a numbered list here to identify the three items

```suggestion
1. The `name` of the resource
   You usually know this going in. In our example above, the name of the resource is `networkSecurityGroups`.

2. The `group` the resource is in 
   This is usually named after the Azure service, for example `microsoft.resources` or `microsoft.documentdb`. In our example entry from above, this is `microsoft.network`.

3. The `api-version` of the resource
   This is usually a date, sometimes with a `-preview` suffix. In our example entry from above, this is `2020-11-01`.
```

File: docs/v2/adding-new-codegenerated-resource.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+## Adding the resource to the code generation configuration file
+The code generation configuration file is located [here](../../hack/generator/azure-arm.yaml). To add a new resource to this file, find the `exportFilters` section of the file and scroll down until you get to a block of `exportFilters` for individual resources. 
+
+Add a new `exportFilter` of kind `include-transitive` at the end of that block, right above this section:
```
Comment (theunrepentantgeek) [inline docs/v2/adding-new-codegenerated-resource.md:31]:
```suggestion
Add a new `exportFilter` of kind `include-transitive` at the end of that block, right _above_ this section:
```

File: docs/v2/adding-new-codegenerated-resource.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  - action: exclude
+    because: We don't want to generate anything else, at the moment.
+```
+
+Your export filter should look like this:
+```yaml
+- action: include-transitive
+  group: <group>
+  version: v*api<api-version with dashes removed>
+  name: <resource name, with trailing "s" removed>
```
Comment (theunrepentantgeek) [inline docs/v2/adding-new-codegenerated-resource.md:43]: Will every resource name become singular just by removing the `s`? 

```suggestion
  name: <singular resource name, typically just remove the trailing "s">
```

**Review by theunrepentantgeek — APPROVED** [review]
This is a really good start.

File: docs/v2/adding-new-codegenerated-resource.md (hunk 5 above)
Comment (matthchr) [reply docs/v2/adding-new-codegenerated-resource.md:43]: Good catch

File: docs/v2/adding-new-codegenerated-resource.md (hunk 2 above)
Comment (matthchr) [reply docs/v2/adding-new-codegenerated-resource.md:17]: Updated

File: docs/v2/adding-new-codegenerated-resource.md (hunk 3 above)
Comment (matthchr) [reply docs/v2/adding-new-codegenerated-resource.md:26]: Done

File: docs/v2/adding-new-codegenerated-resource.md (hunk 4 above)
Comment (matthchr) [reply docs/v2/adding-new-codegenerated-resource.md:31]: Done

File: docs/v2/adding-new-codegenerated-resource.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+If the resource is not in either of the above places and cannot be deployed via an ARM template, then it's not an ARM resource and currently cannot be code generated.
+
+## Determine a resource to add
+There are three key pieces of information required before adding a resource to the code generation configuration file, and each of them can be found in the 
+[ARM template JSON schema](https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json) or the [auto-generated ARM template JSON schema](https://schema.management.azure.com/schemas/common/autogeneratedResources.json). To get started, find the entry for the resource in one of the two templates mentioned. For example, the entry for Azure Network Security Groups looks like this:
+`{ "$ref": "https://schema.management.azure.com/schemas/2020-11-01/Microsoft.Network.json#/resourceDefinitions/networkSecurityGroups" },`
+
+**Note**: In many cases there will be multiple entry's for the same resource, each with a different `api-version`. It is _strongly_ recommended that you use the latest available non-preview `api-version`.
```
Comment (Porges) [inline docs/v2/adding-new-codegenerated-resource.md:17]:
```suggestion
**Note**: In many cases there will be multiple entries for the same resource, each with a different `api-version`. It is _strongly_ recommended that you use the latest available non-preview `api-version`.
```

File: docs/v2/adding-new-codegenerated-resource.md (hunk 1 above)
Comment (matthchr) [reply docs/v2/adding-new-codegenerated-resource.md:63]: Will add this as I encounter them. Last few resources have been so painless that there were no errors.

**Pull request merged**